pkgs/development/libraries/gtk+/3.0-immodules.cache.patch

```diff
@@ -0,0 +1,24 @@
+--- a/gtk/deprecated/gtkrc.c	2016-04-02 18:43:08.401663112 +0900
++++ b/gtk/deprecated/gtkrc.c	2016-04-02 18:29:19.927608592 +0900
+@@ -771,5 +771,20 @@
+   const gchar *var = g_getenv ("GTK_IM_MODULE_FILE");
+   gchar *result = NULL;
+ 
++  // check NIX_PROFILES paths.
++  const gchar *nixProfilesEnv = g_getenv ("NIX_PROFILES");
++  gchar *cachePath;
++  guint i;
++
++  if(nixProfilesEnv){
++    gchar **paths = g_strsplit(nixProfilesEnv, " ", -1);
++    for (i = 0; paths[i] != NULL; i++){
++      cachePath = g_build_filename(paths[i], "etc", "gtk-3.0", "immodules.cache", NULL);
++      if( g_file_test( cachePath, G_FILE_TEST_EXISTS) ){
++        result = g_strdup(cachePath);
++      }
```

Comment:
@gebner Thanks for the memory leak information, I added `g_strfreev(paths);` and `g_free(cachePath);`, is that enough to fix it?

> Aside from that, shouldn't the environment variable GTK_IM_MODULE_FILE take precedence if it is set?

It will, as the check for the `GTK_IM_MODULE_FILE` is after this patch logic and update `result`.

The code before patching is:

```
gchar *
gtk_rc_get_im_module_file (void)
{
  const gchar *var = g_getenv ("GTK_IM_MODULE_FILE");
  gchar *result = NULL;

// patch comes here

  if (var)
    result = g_strdup (var);

  if (!result)
    {
      if (im_module_file)
	result = g_strdup (im_module_file);
      else
        result = gtk_rc_make_default_dir ("immodules.cache");
    }

  return result;
}
```

PS: I have almost no C experience so if you can write a better patch I will be very happy to use it instead :)